### PR TITLE
Feature/fetch quiz list

### PR DESCRIPTION
**Pull request by LorenaL9**
Simulator Screen Shot - iPhone 13 - 2022-08-03 at 10 48 24

### CODE REVIEW

File: QuizApp/Quiz/View/QuizViewModel.swift [hunk 1]
```diff
@@ -3,61 +3,47 @@ import UIKit
...
-        movies.append(QuizModel(category: .movies, name: "Comedy", difficulty: .easy))
-        movies.append(QuizModel(category: .movies, name: "Horror", difficulty: .normal))
-        movies.append(QuizModel(category: .movies, name: "Drama", difficulty: .hard))
-        movies.append(QuizModel(category: .movies, name: "Action", difficulty: .normal))
-        movies.append(QuizModel(category: .movies, name: "Animated", difficulty: .easy))
-
-        history.append(QuizModel(category: .history, name: "Middle Ages", difficulty: .normal))
-
-        quizzes = sport
+    func setQuizzes(category: QuizCategory) {
```
Comment (dorastipkovic) [inline QuizApp/Quiz/View/QuizViewModel.swift:25]: can be private

Comment (dorastipkovic) [inline QuizApp/Quiz/View/QuizViewModel.swift:25]:
```suggestion
    func getQuizzes(for category: QuizCategory) {
```

File: QuizApp/Quiz/View/QuizViewModel.swift [hunk 2]
```diff
@@ -3,61 +3,47 @@ import UIKit
...
+        Task {
+            quizResponse = await quizUseCase.getQuizzes(category: category.text) ?? []
+            quizzes = quizResponse.map { quiz in
+                Quiz(fromModel: quiz, color: category.color)
+            }
```
Comment (dorastipkovic) [inline QuizApp/Quiz/View/QuizViewModel.swift:30]:
```suggestion
            quizzes = quizResponse.map { Quiz(fromModel: $0, color: category.color)  }
```

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

File: QuizApp/Quiz/View/QuizViewModel.swift [hunk 3]
```diff
@@ -3,61 +3,46 @@ import UIKit
...
-        case 3:
-            quizzes = history
-        default:
-            quizzes = []
+        for category in QuizCategory.allCases {
+            let isSelectedSomeSpecificCategory = category.id == selectedIndex
+
+            if isSelectedSomeSpecificCategory {
+                getQuizzes(for: category)
+            }
```
Comment (dorastipkovic) [inline QuizApp/Quiz/View/QuizViewModel.swift:42]:
```suggestion
        let selectedCategory = QuizCategory(rawValue: selectedIndex)
        getQuizzes(for: selectedCategory)
```

smth like this

File: QuizApp/User/View/UserViewController.swift [hunk 4]
```diff
@@ -42,13 +42,15 @@ class UserViewController: UIViewController {
     private func bindViewModel() {
         viewModel
             .$username
+            .receive(on: RunLoop.main)
```
Comment (dorastipkovic) [inline QuizApp/User/View/UserViewController.swift:45]: Properties published from VM should already be on main.

File: QuizApp/Quiz/Client/Models/QuizResponse.swift [hunk 5]
```diff
@@ -0,0 +1,11 @@
+struct QuizResponse: Decodable {
+
+    let category: String
```
Comment (dorastipkovic) [inline QuizApp/Quiz/Client/Models/QuizResponse.swift:3]: category should be a separate model here and in lower layers

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

File: QuizApp/Quiz/Client/Models/QuizResponse.swift (hunk 5 above)
Comment (LorenaL9) [reply QuizApp/Quiz/Client/Models/QuizResponse.swift:3]: This will be fixed in next PR.

**Review by dorastipkovic — APPROVED** [review]

**Pull request merged**